pkg/amphora/client.go

```diff
@@ -55,12 +79,47 @@ type Client struct {
 	HTTPClient http.Client
 }
 
-const secretShareURI = "/intra-vcp/secret-shares"
+const secretShareURI = "/secret-shares"
```

Comment:
Of course I understand why this _public_ interface is used at this point, since only there the corresponding function is provided, but I am still not sure if this should really be the case.

From Amphora Service [documentation](https://github.com/carbynestack/amphora/tree/master/amphora-service)
> Amphora provides three interfaces:
> 
>     User facing: Used by clients for secret sharing, accessing, and managing secret shares and their tags (/input-masks, /masked-input, /secret-shares).
>     Intra VCP: Used by services running in the same Virtual Cloud Provider to access locally stored secret shares, e.g. for ingesting into a computation (/intra-vcp).
>     Inter VCP: Used for communication with other Virtual Cloud Providers in the same Virtual Cloud (/inter-vcp).
> 
>     ⚠️ When setting up the system, it is necessary to limit access to non-public endpoints (/intra-vcp and /inter-vcp) to local services or the services provided by remote Virtual Cloud Providers.
> 

While the _user facing_ and _inter-vcp_ interfaces are supposed to be exposed to external parties and should be protected by authentication (and authorization) in the long run, _intra-vcp_ is considered as vcp internal only, blocked from external access and will eventually not require authentication.

With this, ephemeral would be required to authenticate to the `/secret-shares` endpoint in the future or new mechanisms to be added. 

Could it be more convenient to add similar functionality to the _inter-vcp_ interface, allowing to retrieve a list of secrets matching the specified criteria, but possibly providing only a list of IDs rather than a whole set of metadata that will be discarded anyway?

Reply:
Good point!

This would require changes in Amphora, so do we want to have this PR as-is and then have the additional endpoint(s) in Amphora as 2nd step, or do you see this part of this initial PR?

Also, architectural-wise we probably have another look at what kind of endpoint we'd want to add then?
- An endpoint that returns Secret Meta-Information (whether only the IDs or all MetaInfos) based on Tags Filters and sorted by a tag
- An endpoint that returns Secret Shares based on Tag Filters and sorted by a tag

Since for this usecase latter is probably what would be easier for ephemeral since then it would not need n+1 calls to amphora?
Back when we were planning how to implement this, that was our original idea, but we opted for the way it currently is as it would require changes only in ephemeral and not in amphora as well.

We had internally discussed the options
- Have another party handle the TagFilter -> SecretId conversions and then invoke ephemeral as normal with only the IDs given. Pro: Uses existing Code; Ordering can be made deterministic by the service; Cons: Extra Infrastructure element 
- Have an endpoint in Amphora that offers TagFilter -> [SecretShare...] and call that from ephemeral and feed the results to MP-SPDZ. Pro: Only 1 HTTP Request; Cons: Changes in both Amphora and Ephemeral
- Have a 2-step process in ephemeral: 1st fetch all SecretIDs from Amphora, then load the SecretIDs as normal. Pros: Only changes in Ephemeral -> Fast to implement; Con: N+1 Problem

We decided for the last approach since it was the fastest way for us to go forward.


So, the outstanding question is twofold
- Is our approach for now okay, or should we change the approach to one of the other options?
- Do we want to keep the PR as it is for now and enhance it later?

Reply:
As you said, I see this as an architectural decision rather than a simple improvement, as with the discussion about the sorting parameter. Therefore I'd say this should be addressed as part of this PR / the issue addressed.

> An endpoint that returns Secret Shares based on Tag Filters and sorted by a tag

Seems like the best approach to me as it reduces complexity (number of calls) and logic is implemented where it belongs to. 

With this I think it would be best to create a ticket that addresses the actual purpose of this PR containing two tasks:
1. add an _intra-vcp_  endpoint to amphora to return a sorted list of secret shares matching given criteria 
2. consume the new endpoint in ephemeral as part of this PR